diff --git a/include/parlay/parallel.h b/include/parlay/parallel.h
index 5f32807..39ae320 100644
--- a/include/parlay/parallel.h
+++ b/include/parlay/parallel.h
@@ -131,6 +131,10 @@ static void par_do3_if(bool do_parallel, Lf&& left, Mf&& mid, Rf&& right) {
 #elif defined(PARLAY_TBB)
 #include "internal/scheduler_plugins/tbb.h"         // IWYU pragma: keep, export
 
+// Argobots-based work-stealing runtime
+#elif defined(PARLAY_ARGOBOTS)
+#include "parlay_plugin/argobots.h"             // IWYU pragma: keep, export
+
 // No Parallelism
 #elif defined(PARLAY_SEQUENTIAL)
 #include "internal/scheduler_plugins/sequential.h"  // IWYU pragma: keep, export
